ts/screens/profile/LanguagesPreferencesScreen.tsx

```diff
@@ -49,160 +43,172 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
 /**
  * Allows the user to select one of the available Languages as preferred
  */
-class LanguagesPreferencesScreen extends React.PureComponent<
-  LanguagesPreferencesScreenProps,
-  State
-> {
-  constructor(props: LanguagesPreferencesScreenProps) {
-    super(props);
-    this.state = { isLoading: false };
-  }
-
-  private isAlreadyPreferred = (language: Locales) =>
-    // if the preferred Lanuage is not set, we check if language is the same in use
-    pipe(
-      this.props.preferredLanguage,
-      O.map(l => l === language),
-      O.getOrElse(() => getLocalePrimaryWithFallback() === language)
-    );
-
-  private onLanguageSelected = (language: Locales) => {
-    if (!this.isAlreadyPreferred(language)) {
-      Alert.alert(
-        I18n.t("profile.preferences.list.preferred_language.alert.title") +
-          " " +
-          I18n.t(`locales.${language}` as TranslationKeys) +
-          "?",
-        I18n.t("profile.preferences.list.preferred_language.alert.subtitle"),
-        [
-          {
-            text: I18n.t("global.buttons.cancel"),
-            style: "cancel"
-          },
-          {
-            text: I18n.t("global.buttons.confirm"),
-            style: "default",
-            onPress: () => {
-              this.setState({ selectedLocale: language }, () => {
-                this.props.upsertProfile(language);
-              });
-            }
-          }
-        ],
-        { cancelable: false }
-      );
-    }
-  };
 
-  public componentDidUpdate(prevProps: Readonly<Props>) {
+const LanguagesPreferencesScreen = () => {
+  const viewRef = createRef<View>();
+  const dispatch = useIODispatch();
+  const selectedLanguage = useRef<string | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
+  const profile = useIOSelector(profileSelector, _.isEqual);
+  const prevProfile = usePrevious(profile);
+  const preferredLanguageSelect = useIOSelector(
+    preferredLanguageSelector,
+    _.isEqual
+  );
+  const preferredLanguage = pipe(
+    preferredLanguageSelect,
+    O.getOrElse(() => "it")
+  );
+
+  const upsertProfile = useCallback(
+    (language: Locales) =>
+      dispatch(
+        profileUpsert.request({
+          preferred_languages: [fromLocaleToPreferredLanguage(language)]
+        })
+      ),
+    [dispatch]
+  );
+
+  const preferredLanguageSaveSuccessDispatch = useCallback(
+    (language: Locales) =>
+      dispatch(
+        preferredLanguageSaveSuccess({
+          preferredLanguage: language
+        })
+      ),
+    [dispatch]
+  );
+
+  const renderedItem: Array<RadioItem<string>> = availableTranslations.map(
+    item => ({
+      value: I18n.t(`locales.${item}`, {
+        defaultValue: item
+      }),
+      id: item,
+      techName: `${item}-${item.toUpperCase()}`
+    })
+  );
+
+  const initialSelectedItem = renderedItem.find(
+    item => item.id === preferredLanguage
+  )?.id;
```

Comment:
Maybe this part can be optimized by memoizing the result of these two iterations to avoid unnecessary re-renders. What do you think?
```suggestion
  const renderedItem: Array<RadioItem<string>> = useMemo(() =>
    availableTranslations.map(
      item => ({
        value: I18n.t(`locales.${item}`, {
          defaultValue: item
        }),
        id: item,
        techName: `${item}-${item.toUpperCase()}`
      })
    ), []);

  const initialSelectedItem = useMemo(() =>
    renderedItem.find(
      item => item.id === preferredLanguage
    )?.id, [renderedItem, preferredLanguage]);

```

Reply:
Applied! thanks! 🚀